Source/DafnyLanguageServer/Handlers/DafnyDefinitionHandler.cs

```diff
@@ -29,23 +29,24 @@ public class DafnyDefinitionHandler : DefinitionHandler {
     public override Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken) {
       DafnyDocument? document;
       if(!_documents.TryGetDocument(request.TextDocument, out document)) {
-        _logger.LogWarning("location requested for unloaded document {}", request.TextDocument.Uri);
+        _logger.LogWarning("location requested for unloaded document {DocumentUri}", request.TextDocument.Uri);
         return Task.FromResult(new LocationOrLocationLinks());
       }
       ILocalizableSymbol? symbol;
       if(!document.SymbolTable.TryGetSymbolAt(request.Position, out symbol)) {
-        _logger.LogDebug("no symbol was found at {} in {}", request.Position, request.TextDocument);
+        _logger.LogDebug("no symbol was found at {Position} in {Document}", request.Position, request.TextDocument);
```

Comment:
What's the effect of naming these template variables? This seems to correspond with [the documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions.logdebug?view=dotnet-plat-ext-5.0#microsoft-extensions-logging-loggerextensions-logdebug(microsoft-extensions-logging-ilogger-system-s) but I don't understand the effect.

Reply:
Personally, I was not too fond of naming the placeholders either, having to name the placeholders. But it appears to me that just using `{}` is an undocumented feature that works with the default logging implementation as well as with NLog. Unfortunately, it broke with the switch to Serilog. I'd have preferred to stick to NLog, but it just didn't want to work with the updated LSP backend.
When using the logger solely with file/console outputs, it hardly has any use. In contrast, if you're logging into a database or in a structured format such as JSON, you may store key/value pairs for these placeholders.